src/actions/expressions.js

```diff
@@ -22,21 +26,27 @@ function expressionExists(expressions, input) {
  * @memberof actions/pause
  * @static
  */
-function addExpression(input: string) {
+function addExpression(input: string, { visible = true }: Object = {}) {
   return ({ dispatch, getState }: ThunkArgs) => {
     const expressions = getExpressions(getState());
     if (!input || expressionExists(expressions, input)) {
-      return;
+      const expression = getExpression(getState(), input);
+      if (!expression.visible && visible) {
+        dispatch(deleteExpression(expression));
```

Comment:
hmm, maybe we want to use `await` here which will mean the outer function `return ({ dispatch, getState }: ThunkArgs) => {` should be async. There are examples of this elsewhere